src/openrct2-ui/windows/LoadSave.cpp

```diff
@@ -992,6 +993,8 @@ static void WindowLoadsaveSelect(rct_window* w, const char* path)
             WindowLoadsaveInvokeCallback(MODAL_RESULT_OK, pathBuffer);
             window_close_by_class(WC_LOADSAVE);
             gfx_invalidate_screen();
+            gIsAutosaveLoaded = gIsAutosave;
+            gFirstTimeSaving = false;
```

Comment:
I don't think this is the right place for this code. This dialog gets used as a general file dialog, so it might not actually be loading a game at this point. It should be moved to the load game routine.

It also wouldn't work if you loaded a saved game via command line.

Reply:
Good point, I moved the code over to the load game routine.